param/parameterized.py

```diff
@@ -1089,6 +1141,9 @@ def params(self_, parameter_name=None):
             setattr(cls,'_%s__params'%cls.__name__,paramdict)
             pdict= paramdict
 
+        if self_.self is not None:
+            pdict.update(self_.self._instance__params)
+
```

Comment:
I'm not entirely sure, but it seems that if copies are made (to make a parameter into an instance parameter) in ``__getitem__``,  then the parameter must always exist as a normal parameter first and then as an instance parameter. Won't both those parameters have the same name and therefore won't you have a key clashing in this dictionary?

Reply:
Unless the semantics is updated so `params` always prioritizes returns instance parameters. I'm not sure that would always be a good policy so maybe there should be a new flag to control this?

Reply:
Yes, a flag makes definite sense.